src/localization/online_localizer/online_localizer.cpp

```diff
@@ -125,110 +125,76 @@ void OnlineLocalizer::writeOutExpanded(const std::string &filename) const {
-    printf("[INFO][OnlineLocalizer] NOT LOST\n");
-    while (!_frontier.empty() && !row_reached) {
-      // counterNodes++;
-      Node expandedNode = _frontier.top();
-      _frontier.pop();
+    // printf("[INFO][OnlineLocalizer] NOT LOST\n");
+    while (!frontier_.empty() && !row_reached) {
+      Node expandedNode = frontier_.top();
+      frontier_.pop();
       int expanded_row = expandedNode.quId;
 
       if (!nodeWorthExpanding(expandedNode)) {
         continue;
       }
-      // printf("Node %d %d  %d worth expanding\n", expandedNode.quId,
-      // expandedNode.refKey.refId, expandedNode.refKey.seqId);
       children = successorManager_->getSuccessors(expandedNode);
       updateGraph(expandedNode, children);
       updateSearch(children);
       if (expanded_row == quId - 1) {
         row_reached = true;
       } else if (expanded_row >= quId) {
-        printf(
-            "[ERROR][Online Localizer] You have expanded the nodes higher than "
-            "current query image id. Something went wrong\n");
-        exit(EXIT_FAILURE);
+        LOG(FATAL) << "You have expanded the nodes higher than "
+                      "current query image id. Something went wrong";
       }
     }
   }
   for (const Node &n : children) {
-    _expandedRecently.insert(n);
+    expandedRecently_.insert(n);
   }
 }
 
 void OnlineLocalizer::processImage(int quId) {
-  printf("[DEBUG][OnlineLocalizer] Checking image %d\n", quId);
+  LOG(INFO) << "Checking image " << quId;
   if (quId == 0) {
-    _needReloc = true;
+    needReloc_ = true;
   }
   matchImage(quId);
 
-  // printf("[INFO] Qu %d frontier empty %d\n", qu, frontier.empty());
-  if (_frontier.empty()) {
-    printf("[ERROR] Frontier is empty!\n");
-    exit(EXIT_FAILURE);
-  }
-  // Lost if more than 80% hidden nodes
-  if (isLost(_slidingWindowSize, 0.8)) {
-    _needReloc = true;
-    printf("LOST\n");
-  } else {
-    // not lost anymore
-    _needReloc = false;
-  }
-}
+  CHECK(!frontier_.empty()) << "Frontier is empty! Something bad happened.";
 
-void OnlineLocalizer::run() {
-  if (!isReady()) {
-    printf(
-        "[ERROR][OnlineLocalizer] Online Localizer is not ready to work. Check "
-        "if all needed parameters are set.\n");
-    exit(EXIT_FAILURE);
-  }
-
-  Timer timer;
-  // For the first image consider lost
-  // for every image in the query set
-  for (int qu = 0; qu < _querySize; ++qu) {
-    // while the graph is not expanded till row 'qu'
-    timer.start();
-    processImage(qu);
-    timer.stop();
-    printf("[OnlineLocalizer] Matched image %d\n", qu);
-    timer.print_elapsed_time(TimeExt::MSec);
-    printf("==========================================\n");
-    visualize();
-  }
-  printf("[DEBUG][OnlineLocalizer] Localization finished\n");
-  if (_vis) {
-    _vis->processFinished();
+  // Lost if more than kMaxLostNodesRatio are hidden nodes.
+  if (isLost(kSlidingWindowSize_, kMaxLostNodesRatio)) {
+    needReloc_ = true;
```

Comment:
nit: I usually prefer longer names. So instead of `needReloc_` I would probably use `needRelocalization_`, but this is of course a matter of personal preference. The same holds for other variables in the code.